deploy/docker/server.py

```
@@ -178,6 +197,31 @@ async def add_security_headers(request: Request, call_next):
+
+
+
+def _ensure_directory(path: str) -> None:
+    os.makedirs(path, mode=0o750, exist_ok=True)
+
+
+# Legacy function - replaced by protocol-based error handling
+# Kept for backward compatibility during transition
+def _extract_user_friendly_error(error_msg: str, operation: str = "operation") -> str:
+    """Legacy function - use error_handler.parse_network_error() instead."""
+    from error_handler import parse_network_error, handle_error
```

Ruff (0.13.1)

Redefinition of unused `parse_network_error` from line 69

Remove definition: `parse_network_error`

(F811)